### DISCUSSION TITLE
import yarp Error, or do i need to rebuild the yarp with pythonsupport ?

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#633] [Q&A] [answered]
Sorry to ask a repeated question here, as i asked somewhere else https://github.com/robotology/yarp/discussions/2975 , the question is 

I set the PYTHONPATH, but still failed to import yarp.

> (base) icub@icubsrv:/usr/local/src/robot/robotology-superbuild/build/src/YARP/bindings$ echo $PYTHONPATH
/usr/local/src/robot/robotology-superbuild/build/src/YARP:/usr/local/src/robot/robotology-superbuild/build/src/YARP/bindings
(base) icub@icubsrv:/usr/local/src/robot/robotology-superbuild/build/src/YARP/bindings$ python
Python 3.8.16 (default, Mar  2 2023, 03:21:46) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
\>>> import yarp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'yarp

Or if i need to rebuild or super-build the yarp with  pythonsupport, how to do it?

Regards.

**Comment by pattacini** [comment, marked as answer]
>Or if i need to rebuild or super-build the yarp with pythonsupport, how to do it?

Yes, you ought to rebuild the superbuild with Python enabled. To this end, there's the following CMake flag to toggle on:
- `ROBOTOLOGY_USES_PYTHON`

See also https://github.com/robotology/robotology-superbuild/blob/master/README.md?plain=1#L141-L147.

cc @traversaro
